mobile/src/main/java/com/truckmuncher/app/customer/TruckDetailsActivity.java

```diff
@@ -95,14 +95,7 @@ protected void onCreate(Bundle savedInstanceState) {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case android.R.id.home:
-                Intent parent = NavUtils.getParentActivityIntent(this);
-                if (NavUtils.shouldUpRecreateTask(this, parent)) {
-                    NavUtils.navigateUpTo(this, parent);
-                } else {
-                    TaskStackBuilder.create(this)
-                            .addParentStack(this)
-                            .startActivities();
-                }
```

Comment:
Removing this code breaks up navigation when deep linking into the app. Perhaps we need to think about what the correct up hierarchy is though.